File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectPersistance/src/main/java/ro/mpp2025/javaprojectui/database/ParticipantRepository.java
//package ro.mpp2025.javaprojectui.database;
//
//import org.apache.logging.log4j.LogManager;
//import org.apache.logging.log4j.Logger;
//import ro.mpp2025.javaprojectui.AbstractRepository;
//import ro.mpp2025.javaprojectui.ParticipantORM;
//import ro.mpp2025.javaprojectui.interfaces.IParticipantRepository;
//import ro.mpp2025.javaprojectui.utils.JdbcUtils;
//
//import java.sql.Connection;
//import java.sql.PreparedStatement;
//import java.sql.ResultSet;
//import java.sql.SQLException;
//
//public class ParticipantRepository extends AbstractRepository<Integer, ParticipantORM> implements IParticipantRepository {
//
//    private static final Logger logger= LogManager.getLogger(ParticipantRepository.class);
//
//    private final Connection connection;
//
//    public ParticipantRepository() throws SQLException {
//        logger.info("Initializing ParticipantRepository ");
//        this.connection = JdbcUtils.getInstance().getConnection();
//    }
//
//    @Override
//    protected PreparedStatement findOneQuery(Integer id) throws SQLException {
//        String query = "SELECT * FROM participanti WHERE id = ?";
//        PreparedStatement statement = connection.prepareStatement(query);
//        statement.setObject(1, id);
//
//        return statement;
//    }
//
//    @Override
//    protected PreparedStatement findAllQuery() throws SQLException {
//        String query = "SELECT * FROM participanti";
//        return connection.prepareStatement(query);
//    }
//
//    @Override
//    protected PreparedStatement saveQuery(ParticipantORM entity) throws SQLException {
//        String query = "INSERT INTO participanti(id, nume, cnp) VALUES (?, ?, ?)";
//        PreparedStatement statement = connection.prepareStatement(query);
//        statement.setObject(1, entity.getId());
//        statement.setString(2, entity.getNume());
//        statement.setString(3, entity.getCnp());
//        return statement;
//    }
//
//    @Override
//    protected ParticipantORM buildEntity(ResultSet resultSet) throws SQLException {
//        Integer id = resultSet.getInt("id");
//        String nume = resultSet.getString("nume");
//        String cnp = resultSet.getString("cnp");
//
//        ParticipantORM participant = new ParticipantORM(nume, cnp);
//        participant.setId(id);
//
//        return participant;
//    }
//
//}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Persistence/database/ParticipantRepository.cs
using System.Data;
using Model;
using log4net;

namespace Persistence.database;
public class ParticipantRepository : AbstractRepository<int, Participant>, IParticipantRepo
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(OrganizatorRepository));

        private readonly IDbConnection connection;

        public ParticipantRepository()
        {
            logger.Info("Initializing ParticipantRepository...");
            connection = JdbcUtils.GetInstance().GetConnection();
        }

        protected override IDbCommand FindOneQuery(int id)
        {
            string query = "SELECT * FROM participanti WHERE part_id = @id";
            var command = connection.CreateCommand();
            command.CommandText = query;
            var param = command.CreateParameter();
            param.ParameterName = "@id";
            param.Value = id;
            command.Parameters.Add(param);
            return command;
        }

        protected override IDbCommand FindAllQuery()
        {
            string query = "SELECT * FROM participanti";
            var command = connection.CreateCommand();
            command.CommandText = query;
            return command;
        }

        protected override IDbCommand SaveQuery(Participant entity)
        {
            string query = "INSERT INTO participanti(name, cnp) VALUES ( @name, @cnp)";
            var command = connection.CreateCommand();
            command.CommandText = query;

            var paramNume = command.CreateParameter();
            paramNume.ParameterName = "@name";
            paramNume.Value = entity.Nume;
            command.Parameters.Add(paramNume);

            var paramCnp = command.CreateParameter();
            paramCnp.ParameterName = "@cnp";
            paramCnp.Value = entity.Cnp;
            command.Parameters.Add(paramCnp);

            return command;
        }

        protected override Participant BuildEntity(IDataReader reader)
        {
            int id = reader.GetInt32(reader.GetOrdinal("part_id"));
            string nume = reader.GetString(reader.GetOrdinal("name"));
            string cnp = reader.GetString(reader.GetOrdinal("cnp"));

            Participant participant = new Participant(nume, cnp);
            participant.Id = id;

            return participant;
        }
        
        public new Participant Save(Participant participant)
        {
            using (var command = SaveQuery(participant))
            {
                command.ExecuteNonQuery(); 
                using (var idCommand = connection.CreateCommand())
                {
                    idCommand.CommandText = "SELECT last_insert_rowid();";
                    var id = (long)idCommand.ExecuteScalar();
                    participant.Id = (int)id; 
                }
        
            }
            return participant;
        }
        

    }